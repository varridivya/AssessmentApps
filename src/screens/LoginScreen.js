/* eslint-disable prettier/prettier */

import React from 'react';
import {
    View,
    Text,
    StyleSheet,
    TextInput,
    TouchableOpacity,
} from 'react-native';
import { useSelector, useDispatch } from 'react-redux';

import { setName } from '../redux';
// Native Module
import { NativeModules } from 'react-native';
import { appOnEmulator, LOGIN } from '../constants/constants';

// @flow
const LoginScreen = ({ navigation }) => {

    const { name } = useSelector(state => state.userReducer);
    const dispatch = useDispatch();

    const { DeviceModule } = NativeModules;
    const isEmulator = DeviceModule.isEmulator();
    return (
        <View style={styles.container}>
            {isEmulator && (
                <Text style={styles.yellowText}>{appOnEmulator}</Text>
            )}
            <Text style={styles.headingText}>{LOGIN}</Text>
            <TextInput
                placeholder="Enter Name"
                style={styles.inputView}
                onChangeText={(value) => dispatch(setName(value))}
                value={name}
            />
            <TouchableOpacity onPress={() => navigation.navigate('ScreenTwo')} style={styles.loginBtn}>
                <Text style={styles.loginText}>{LOGIN}</Text>
            </TouchableOpacity>
        </View>
    );
};
const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    loginBtn: {
        width: '50%',
        backgroundColor: 'green',
        borderRadius: 25,
        height: 50,
        alignItems: 'center',
        justifyContent: 'center',
        margin: 40,
    },
    loginText: {
        textAlign: 'center',
        fontSize: 16,
        color: 'white',
    },
    inputView: {
        width: '80%',
        backgroundColor: '#fff',
        borderRadius: 9,
        borderColor: 'black',
        borderWidth: 2,
        height: '10%',
        justifyContent: 'center',
        padding: 20,
    },
    headingText: {
        padding: 20,
        fontSize: 24,
        fontWeight: 'bold',
    },
});
export { LoginScreen };
