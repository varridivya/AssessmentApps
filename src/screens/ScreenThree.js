import React from 'react';
import {
    View,
    Text,
    StyleSheet,
    TouchableOpacity
} from 'react-native';
import { useSelector } from 'react-redux';
import { Go_BACK_TO_HOME_TEXT, HOME } from '../constants/constants';

const ScreenThree = ({ navigation }) => {
    const { name } = useSelector(state => state.userReducer);
    return (
        <View style={styles.root}>
            <View>
                <Text style={styles.nameStyle}>Hi, {name}</Text>
            </View>
            <View style={styles.container}>
                <Text style={{ fontSize: 18 }}>{Go_BACK_TO_HOME_TEXT}</Text>
                <TouchableOpacity onPress={() => navigation.navigate('Home')} style={styles.loginBtn}>
                    <Text style={styles.loginText}>{HOME}</Text>
                </TouchableOpacity>
            </View>
        </View>
    );
}
const styles = StyleSheet.create({
    root: {
        flex: 1,
    },
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    nameStyle: { fontSize: 24, fontWeight: 'bold', textAlign: 'right', margin: 10 },
    loginBtn: {
        width: '50%',
        backgroundColor: '#009688',
        borderRadius: 25,
        height: 50,
        alignItems: 'center',
        justifyContent: 'center',
        margin: 40,
    },
    loginText: {
        textAlign: 'center',
        fontSize: 16,
        color: 'white',
    },
});
export { ScreenThree };