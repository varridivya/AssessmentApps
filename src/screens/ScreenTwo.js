/* eslint-disable prettier/prettier */
import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import SwipeButton from 'rn-swipe-button';
import { useSelector } from 'react-redux';
import thumbIcon from '../images/thumbIcon.png';
import Buttons from '../components/Buttons';
import PropTypes from 'prop-types';

const ScreenTwo = (props) => {
    const { name } = useSelector(state => state.userReducer);
    return (
        <View style={styles.container}>
            <View>
                <Text style={styles.nameStyle}>Hi, {name}</Text>
            </View>
            <View style={styles.container1}>
                <Buttons onPress={() => props.navigation.navigate('ScreenThree')} style={styles.appButtonContainer} />
                <Buttons style={[styles.appButtonContainer, { backgroundColor: 'grey' }]} onPress={() => props.navigation.navigate('ScreenThree')} />
                <Buttons style={[styles.appButtonContainer, { backgroundColor: '#007bff' }]} onPress={() => props.navigation.navigate('ScreenThree')} />
                <SwipeButton
                    containerStyles={{ borderRadius: 8 }}
                    disabled={false}
                    swipeSuccessThreshold={70}
                    height={45}
                    width={325}
                    title="Slide to continue"
                    onSwipeSuccess={() => props.navigation.navigate('ScreenThree')}
                    railBackgroundColor="white"
                    railStyles={{ borderRadius: 5 }}
                    thumbIconImageSource={thumbIcon}
                    thumbIconStyles={{ borderRadius: 5, backgroundColor: 'black' }}
                />
            </View>
        </View>
    );
};
ScreenTwo.props = {
    navigation: PropTypes.any,
};
const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: 'rgb(35,33,49)',
    },
    container1: {
        flex: 1,
        paddingTop: '85%',
        paddingLeft: 30,
        paddingRight: 30,
        backgroundColor: 'rgb(35,33,49)',
    },
    appButtonContainer: {
        borderRadius: 10,
        paddingVertical: 10,
        paddingHorizontal: 12,
        marginVertical: 10,
    },
    appButtonText: {
        fontSize: 18,
        color: '#fff',
        fontWeight: 'bold',
        alignSelf: 'center',
        textTransform: 'uppercase',
    },
    root: {
        width: '100%',
        height: '100%',
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
    },
    nameStyle: {
        fontSize: 24,
        fontWeight: 'bold',
        textAlign: 'right',
        margin: 10,
        color: 'yellow',
    },
});
export { ScreenTwo };
