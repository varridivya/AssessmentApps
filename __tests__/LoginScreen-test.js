import 'react-native';
import React from 'react';
import { Button } from 'react-native';
import { configure, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { LoginScreen } from '../src/screens/LoginScreen';
import renderer from 'react-test-renderer';
import { useSelector, useDispatch } from "react-redux";

configure({ adapter: new Adapter() });

jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn(),
    useDispatch: jest.fn(),
}));

describe("MyComponent", () => {
    beforeAll(() => {
        useSelector.mockImplementation(() => ({
            name: 'Divya',
        }));
    });
    it('should render a placeholder', () => {
        const placeholder_text = 'Enter name';
        const wrapper = shallow(<LoginScreen placeholder={placeholder_text} />);
        expect(wrapper.prop('placeholder')).toEqual(placeholder_text);
    });
    it('renders correctly', () => {
        const wrapper = shallow(<LoginScreen />);
        expect(wrapper).toMatchSnapshot();
    });

    it('to match snapshot for native version', () => {
        const tree = renderer.create(<LoginScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('Button  test', () => {
        const component = shallow(<LoginScreen />);
        const refreshTrigger = component.find(Button).at(0);
        refreshTrigger.props('onPress')();
        expect(component).toMatchSnapshot();
    });
});