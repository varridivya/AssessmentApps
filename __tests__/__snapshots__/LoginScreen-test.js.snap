// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent renders correctly 1`] = `ShallowWrapper {}`;

exports[`MyComponent to match snapshot for native version 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    You are running the app on emulator
  </Text>
  <Text
    style={
      Object {
        "fontSize": 24,
        "fontWeight": "bold",
        "padding": 20,
      }
    }
  >
    LOGIN
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter Name"
    style={
      Object {
        "backgroundColor": "#fff",
        "borderColor": "black",
        "borderRadius": 9,
        "borderWidth": 2,
        "height": "10%",
        "justifyContent": "center",
        "padding": 20,
        "width": "80%",
      }
    }
    value="Divya"
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "green",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "margin": 40,
        "opacity": 1,
        "width": "50%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      LOGIN
    </Text>
  </View>
</View>
`;
